eth/protocols/snap/sync.go

```diff
@@ -1786,10 +1817,13 @@ func (s *Syncer) processStorageResponse(res *storageResponse) {
 							// Make sure we don't overflow if the step is not a proper divisor
 							last = common.HexToHash("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff")
 						}
+						batch := s.db.NewBatch()
 						tasks = append(tasks, &storageTask{
-							Next: next,
-							Last: last,
-							root: acc.Root,
+							Next:     next,
```

Comment:
Wouldn't it be better to use `next` from `res.hashes[i]` ( or wherever the last returned key is?) instead of fetching the storage trie from `0000`?